recipes/full_finetune_distributed.py

```diff
@@ -201,13 +201,19 @@ def load_checkpoint(self, cfg_checkpointer: DictConfig) -> Dict[str, Any]:
         Extract the checkpoint state from file and validate. If resume_from_checkpoint
         is True, this also includes the recipe state.
         """
+        # If async checkpointing is enabled, set the resume from checkpoint as false for the checkpointer.
+        # We will use DistributedCheckpointer instead to load the checkpoint.
+        resume_from_checkpoint: bool = (
+            False if self._enable_async_checkpointing else self._resume_from_checkpoint
+        )
```

Comment:
I'm a bit confused by this.. is it just because of the if statement for `self._update_recipe_state` (i.e. we do not need to explicitly load state when using DCP)? If so I think we should just use a different variable e.g. `should_load_recipe_state = self._resume_from_checkpoint and not self._enable_async_checkpointing` or something to be more explicit.

Reply:
Yes. Thats right. I agree. Current way is a bit sloppy to avoid loading the recipe state. I will update the variables to make it a bit more readable.